pandas/tests/test_strings.py

```diff
@@ -469,6 +469,62 @@ def test_replace_callable(self):
         exp = Series(['bAR', NA])
         tm.assert_series_equal(result, exp)
 
+    def test_replace_compiled_regex(self):
+        # GH 15446
+        values = Series(['fooBAD__barBAD', NA])
+
+        # test with compiled regex
+        pat = re.compile(r'BAD[_]*')
+        result = values.str.replace(pat, '')
+        exp = Series(['foobar', NA])
+        tm.assert_series_equal(result, exp)
```

Comment:
so looks like this is failing of the builds: https://circleci.com/gh/pandas-dev/pandas/188?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

This is a build where we override LOCALE='C' You can tests locally by adding 

```
import locale
locale.setlocale(locale.LC_ALL, 'C')
```

at the top of ``pandas/pandas/__init__.py``

and running with 3.5

not entirely sure what is happening